src/main/scala/cromwell/authentication/oauth2/google/GoogleCredentialFactory.scala

```diff
@@ -9,25 +9,38 @@ import com.google.api.client.googleapis.auth.oauth2.{GoogleAuthorizationCodeFlow
 import com.google.api.client.googleapis.extensions.java6.auth.oauth2.GooglePromptReceiver
 import com.google.api.client.http.HttpTransport
 import com.google.api.client.json.JsonFactory
+import com.google.api.client.json.jackson2.JacksonFactory
 import com.google.api.client.util.store.FileDataStoreFactory
 import com.typesafe.config.{Config, ConfigFactory}
+import cromwell.util.google.GoogleScopes
+import org.slf4j.LoggerFactory
 
 import scala.collection.JavaConverters._
 
 object GoogleCredentialFactory {
   private lazy val GoogleConf = ConfigFactory.load.getConfig("google")
   private lazy val GoogleAuthScheme = GoogleConf.getString("authScheme").toLowerCase
+  private lazy val log = LoggerFactory.getLogger("GoogleCredentialFactory")
 
   lazy val from: (JsonFactory, HttpTransport) => Credential = GoogleAuthScheme match {
     case "user" => forUser(GoogleConf.getConfig("userAuth"))
+    case "access" => forAccessToken(GoogleConf.getConfig("accessTokenAuth"))
+    case "refresh" => forRefreshToken(GoogleConf.getConfig("refreshTokenAuth"))
     case "service" => forServiceAccount(GoogleConf.getConfig("serviceAuth"))
   }
 
+  private def filePathToSecrets(secrets: String, jsonFactory: JsonFactory) = {
+    val secretsPath = Paths.get(secrets)
+    if(!secretsPath.toFile.exists() || !secretsPath.toFile.canRead) {
```

Comment:
I don't think the `!secretsPath.toFile.exists()` is necessary as this is written, I suspect `!secretsPath.toFile.canRead` would fail if the file didn't exist.  You could do two cases if you like.